src/explanations/units_variables.yaml

```diff
@@ -171,3 +171,271 @@
       ```
     The values of all members of msg can change for every external function call. In solidity `block.timestamp` and `blockhash` are sources of randomness are not secure. Timestamp and the blockhash can be influenced by the miners.
 
+- title: ABI Encoding and Decoding functions.
+  shortTitle: Encoding/Decoding
+  key: "encoding-decoding"
+  details: |
+    In the context of computer science, an ABI (Application Binary Interface) is an interface between two programme modules, most commonly between operating systems and user programmes. ABI, like API, defines the methods and structures used to interact with the binary contract at a lower level. The ABI instructs the function caller to encode the necessary information, such as function signatures and variable declarations, in a format that the EVM can understand in order to call that function in bytecode; this is known as ABI encoding.
+    
+    To send data to the contract, we need to send it in a way that the contract can read it. That is, they need to be encoded. Solidity has  global variable i.e, abi that has an encode and decode method, so we can use it to encode the parameters of any function. 
+
```

Comment:
The encode/dicode method on the `abi` global variable in Solidity lets us easily encode/decode the parameters of any function.